[doc] Update feature documentation with new behaviours (#742)
Enalean/tuleap-documentation-en: languages/en/user-guide/code-versioning/gitlab.rst

When you click on this reference, you will be redirected to your GitLab instance, on the page displaying the commit tagged with ``<tag_name>``.

Close Tuleap artifact with GitLab commit messages
`````````````````````````````````````````````````

When integrating a GitLab repository in a Tuleap project, there is an option (disabled by default) that allow to close artifacts in this project based on GitLab commit messages.

> If a commit message containing a closure keyword is pushed in a GitLab repository integrated in a Tuleap project and this integration has activated this feature, 

Reviewer: I suggest we use the terms "closing pattern" to match what the [GitLab documentation is using](https://docs.gitlab.com/ee/user/project/issues/managing_issues.html#closing-issues-automatically) in this section. It's likely to cause less confusion to GitLab users and it avoid the usage of a mostly UK word (the US equivalent is cloture).

Author: Thanks for the suggestion, I will modify the content

then the artifact will be closed :

* If the artifact is in the same project of the integration

> * If a :ref:`status semantic <status-semantic>` is defined for this artfact

Reviewer: Do not we need a Done semantic?

Author: I forgot to add this, thanks.

`````````````````````````````````````````````````

When integrating a GitLab repository in a Tuleap project, there is an option (disabled by default) that allow to close artifacts in this project based on GitLab commit messages.
If a commit message containing a closing pattern is pushed in a GitLab repository integrated in a Tuleap project and this integration has activated this feature, 
then the artifact will be closed :

* If the artifact is in the same project of the integration

> * If a :ref:`done semantic <done-semantic>` or a :ref:`status semantic <status-semantic>` is defined for this artfact

Author: Done

* If the artifact is not yet closed
* If the push occurs in the GitLab repository's default branch

> The handled closure keywords (case insensitive) in closing patterns are:

Reviewer: ```suggestion
The following keywords (case insensitive) can be used to close an artifact:
```

Author: Done

* ``Closing``/``Fixing``/``Resolving``/``Implementing``

The artifact will be closed by a Tuleap bot named ``Tracker Workflow Manager`` and a comment will be added to add context about this closure.

> The first done value retrieved by Tuleap will be used in this closure. If there is no done semantic defined, then the first closed value will be used.

Reviewer: ```suggestion
The first done value retrieved by Tuleap will be used to close the artifact. If there is no done semantic defined, then the first closed value will be used.
```

Author: Done
